tests/common/include/tests/common/check_simulation.inl

```diff
@@ -174,4 +175,145 @@ TEST(check_simulation, toy_geometry) {
         EXPECT_NEAR((std::sqrt(var1) - smearer.stddev[1]) / smearer.stddev[1],
                     0, 0.1);
     }
-}
\ No newline at end of file
+}
+
+// Test parameters: <initial momentum, theta direction>
+class TelescopeDetectorSimulation
+    : public ::testing::TestWithParam<std::tuple<scalar, scalar>> {};
+
+TEST_P(TelescopeDetectorSimulation, telescope_detector_simulation) {
+
+    // Create geometry
+    vecmem::host_memory_resource host_mr;
+
+    // Build from given module positions
+    detail::ray<transform3> traj{{0, 0, 0}, 0, {0, 0, 1}, -1};
+    std::vector<scalar> positions = {0.,   50., 100., 150., 200., 250.,
+                                     300., 350, 400,  450., 500.};
+
+    const auto mat = silicon_tml<scalar>();
+    const scalar thickness = 0.17 * unit_constants::cm;
+
+    // Detector type
+    using detector_type =
+        detray::detector<detray::detector_registry::telescope_detector,
+                         covfie::field>;
+
+    // Create B field
+    const vector3 B{0, 0, 2 * unit_constants::T};
+    using b_field_t = typename detector_type::bfield_type;
+
+    const auto detector = create_telescope_detector(
+        host_mr,
+        b_field_t(b_field_t::backend_t::configuration_t{B[0], B[1], B[2]}),
+        positions, traj, 100000. * unit_constants::mm,
```

Comment:
Why do you use rectangles of this size, when you can also try unmasked surfaces? Does the parameter transport not work for ```unmasked```?

Reply:
It will work with `unmasked`. I just prefer `rectangle` to `unmasked`..